### PR TITLE
#431 issue fix

### PR DESCRIPTION
**Pull request by even1024**
Removing references from BaseMolecule class because they don't get copied with shallow copy.

### CODE REVIEW

File: api/c/indigo/src/indigo_layout.cpp [hunk 1]
```diff
@@ -70,8 +70,8 @@ CEXPORT int indigoLayout(int object)
                 mol->clearBondDirections();
                 try
                 {
-                    mol->stereocenters.markBonds();
-                    mol->allene_stereo.markBonds();
+                    mol->stereocentersMarkBonds();
+                    mol->allene_stereoMarkBonds();
```
Comment (mkviatkovskii) [inline api/c/indigo/src/indigo_layout.cpp:74]: I suggest using the same naming style, `alleneStereoMarkBonds`

File: api/c/indigo/src/indigo_misc.cpp [hunk 2]
```diff
@@ -1032,7 +1032,7 @@ CEXPORT int indigoNormalize(int structure, const char* options)
         {
             // Validate cs-trans because it can disappear
             // For example: [O-]/[N+](=C\C1C=CC=CC=1)/C1C=CC=CC=1
-            mol.cis_trans.validate();
+            mol.cis_transValidate();
```
Comment (mkviatkovskii) [inline api/c/indigo/src/indigo_misc.cpp:1035]: Same here, like `validateCisTrans`

File: api/c/tests/unit/basic/deco_sdf.cpp [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        testScaffold(dataPath("molecules/basic/thiazolidines.sdf"), "approx", false);
+        testScaffold(dataPath("molecules/basic/thiazolidines.sdf"), "approx 3", false);
+        testScaffold(dataPath("molecules/basic/sugars.sdf"), "exact", true);
+        testScaffold(dataPath("molecules/basic/sugars.sdf"), "approx", false);
+    }
+    catch (Exception& e)
+    {
+        ASSERT_STREQ("", e.message());
+    }
+}
```
Comment (mkviatkovskii) [inline api/c/tests/unit/basic/deco_sdf.cpp:140]: I don't think we need this test here - C++ unittests should be small and fast. We have this test in integration tests.

File: api/c/indigo/src/indigo_misc.cpp (hunk 2 above)
Comment (even1024) [reply api/c/indigo/src/indigo_misc.cpp]: fixed

File: api/c/tests/unit/basic/deco_sdf.cpp (hunk 3 above)
Comment (even1024) [reply api/c/tests/unit/basic/deco_sdf.cpp:140]: test removed

File: api/c/indigo/src/indigo_layout.cpp (hunk 1 above)
Comment (even1024) [reply api/c/indigo/src/indigo_layout.cpp]: fixed

**Pull request closed without merging**